django/views/generic/detail.py

```diff
@@ -170,7 +173,11 @@ def get_template_names(self):
             # If we still haven't managed to find any template names, we should
             # re-raise the ImproperlyConfigured to alert the user.
             if not names:
-                raise
+                raise ImproperlyConfigured(
+                    "SingleObjectTemplateResponseMixin requires a definition "
+                    "of 'template_name', 'template_name_field', or 'model', "
```

Comment:
I think the correct punctuation is to make the last comma a semicolon since there are already commas in the first part of the "or".